### PR TITLE
Allow hist file into trigger rate application

### LINKED ISSUE #934: Allow the trigger rate application to take the histogram file .hdata as input (and not only .simtel)
This is specially useful to access the additional histograms that are available in the hdata file, for example potentially the histograms for individual telescopes (to be checked)

### PR DESCRIPTION
**Pull request by VictorBarbosaMartins**
In this PR, I implemented some improvements to the trigger rate tool that allows it to calculate the trigger rate for a given an input .hdata.zst file (and not only an .simtel.zst file as it is in main now). However, the configuration used for the simulation is not propagated to the .hdata.zst as it is to the .simtel.zst, which means we do not have the view cone and the energy range used, which are both necessary for the operation of this tool. Therefore, if passing a .hdata to the tool, `--energy_range` and `--view_cone` also have to be passed. One can test this with the new integration test `simtools-calculate-trigger-rate --config tests/integration_tests/config/calculate_trigger_rate_one_file_hdata.yml`. Not passing `--energy_range` or `--view_cone` for an input .hdata file will lead to error, which is tested in the unit tests.

I believe the low coverage by sonarqube is due to the changes in the application itself, which we do not test with unit tests at the moment, so I suggest to ignore it.

Closes #934.

### CODE REVIEW

File: simtools/applications/calculate_trigger_rate.py
Comment (GernotMaier) [inline simtools/applications/calculate_trigger_rate.py:106]: Type = 'str', and help requests floats? Not clear if this is consistent.

File: simtools/applications/calculate_trigger_rate.py
Comment (GernotMaier) [inline simtools/applications/calculate_trigger_rate.py:112]: Maybe 'View cone radius (deg)' (I assume it is the radius, or?)

File: simtools/applications/calculate_trigger_rate.py
Comment (GernotMaier) [inline simtools/applications/calculate_trigger_rate.py:131]: 'derived' --> 'defined'

File: simtools/simtel/simtel_io_histogram.py
Comment (GernotMaier) [inline simtools/simtel/simtel_io_histogram.py:55]: Important: CORSIKA usually works in GeV. Are you sure about the TeV here?

File: simtools/simtel/simtel_io_histogram.py
Comment (GernotMaier) [inline simtools/simtel/simtel_io_histogram.py:90]: I would find this clearer with `self.view_cone = self._set_view_cone(viewcone)`: in the current implementation, the first assignment is immediately overwritten by the follow up statement.

Same comment for the energy range in lines 93/94.

File: simtools/simtel/simtel_io_histogram.py
Comment (GernotMaier) [inline simtools/simtel/simtel_io_histogram.py:234]: `a list of cone` is unclear. "Minimum and maximum of the viewcone radius in deg"

File: simtools/simtel/simtel_io_histogram.py
Comment (GernotMaier) [inline simtools/simtel/simtel_io_histogram.py:306]: Same here, 'list of energies' is not what we want - we want min/max of the energy range.

File: simtools/simtel/simtel_io_histograms.py
Comment (GernotMaier) [inline simtools/simtel/simtel_io_histograms.py:55]: Same comment here. GeV in Corsika?

**Review by GernotMaier — CHANGES_REQUESTED** [review]
Thanks for this. See my minor comments below.

File: simtools/applications/calculate_trigger_rate.py
Comment (VictorBarbosaMartins) [reply simtools/applications/calculate_trigger_rate.py:106]: This is a good point. For some reason the configurator did not manage to add a parameter, which is a list of two floats, so I decided to get a list of str and transform to float in the code. If you happen to know the reason why type=float does not work here, please me know.

File: simtools/applications/calculate_trigger_rate.py
Comment (VictorBarbosaMartins) [reply simtools/applications/calculate_trigger_rate.py]: True. I adjusted the text.

File: simtools/applications/calculate_trigger_rate.py
Comment (VictorBarbosaMartins) [reply simtools/applications/calculate_trigger_rate.py]: Done.

File: simtools/simtel/simtel_io_histogram.py
Comment (VictorBarbosaMartins) [reply simtools/simtel/simtel_io_histogram.py]: Good point. The docstring is confusing, hence I removed the parentheses. But the code is correct. E_range is already given in TeV (this is dealt with in pyeventio).

File: simtools/simtel/simtel_io_histogram.py
Comment (VictorBarbosaMartins) [reply simtools/simtel/simtel_io_histogram.py]: I understand your point. I improved it now.

File: simtools/simtel/simtel_io_histogram.py
Comment (VictorBarbosaMartins) [reply simtools/simtel/simtel_io_histogram.py]: Done.

File: simtools/simtel/simtel_io_histogram.py
Comment (VictorBarbosaMartins) [reply simtools/simtel/simtel_io_histogram.py]: Done.

File: simtools/simtel/simtel_io_histograms.py
Comment (VictorBarbosaMartins) [reply simtools/simtel/simtel_io_histograms.py:55]: As pointed out before, yes, this is correct. We are not directly dealing with CORSIKA parameters here (they are already dealt with in eventio).

**Comment by VictorBarbosaMartins** [comment]
Hi @GernotMaier, thanks for the comments. I think I addressed all of them. Please let me know what to do in regard to the type=str or type=float. Also please have a look again once tests pass. If they fail, I will deal with them later.

File: simtools/applications/calculate_trigger_rate.py
Comment (GernotMaier) [reply simtools/applications/calculate_trigger_rate.py:106]: Don't know. This probably requires to look a bit deeper into argparse.

File: simtools/simtel/simtel_io_histograms.py
Comment (GernotMaier) [reply simtools/simtel/simtel_io_histograms.py:55]: Please remove then "as in CORSIKA configuration", as this is not correct (CORSIKA is strictly in GeV).

**Review by GernotMaier — APPROVED** [review]
Thanks for this.

Please correct the one bit on the energy / CORSIKA configuration. Otherwise ready to merge.

File: simtools/applications/calculate_trigger_rate.py
Comment (VictorBarbosaMartins) [reply simtools/applications/calculate_trigger_rate.py:106]: Yes, definitely not for this PR. I think it is fine as it is but if we really want to understand the reason for it we would have to look into argparse. I opened an issue for that #1005.

**Comment by VictorBarbosaMartins** [comment]
Thanks again @GernotMaier. Will wait until the tests pass again to merge.

**Pull request merged**
